### PR TITLE
Fix user cleanup after tests finish 

### PR DESCRIPTION
**Pull request by amrita-shrestha**
### Description
This PR refactor code so that only successfully created user added on createdUsers array.
Refactor error message 

### Related issue
[Users/groups aren't being cleaned up properly after tests run in OCIS](https://github.com/owncloud/owncloud-test-middleware/issues/120)

### CODE REVIEW

**Review by SwikritiT — APPROVED** [review]
The change looks good to me. Make sure to create a new tag and bump version in web and ocis as well

File: src/helpers/graphHelper.js [hunk 1]
```diff
@@ -21,10 +21,15 @@ exports.createUser = function (
...
-    userSettings.addUserToCreatedUsersList(user, password, displayName, email)
+
     return httpHelper
         .postGraph('users', 'admin', body)
-        .then(res => httpHelper.checkStatus(res, 'Failed while creating user'))
+        .then((res) => {
+            if (res.ok) {
+                userSettings.addUserToCreatedUsersList(user, password, displayName, email)
+            }
+            httpHelper.checkStatus(res, 'Failed while creating user')
```
Comment (kiranparajuli589) [inline src/helpers/graphHelper.js:31]: maybe adding to created list after this check if better. status code also should be good before push

**Review by kiranparajuli589 — COMMENTED** [review]
- In the issue `401` i.e. unauthorized is mentioned in the response status while cleaning in the after scenario. how does this change solves that case :thinking: 
- Still for a successful scenario like you mentioned here https://user-images.githubusercontent.com/54478846/182773375-1063ef25-5464-4d3a-970b-43375432ed80.webm. For this, the code was pushing the users to the created list right? There is no fail in the create request. so, I'm surprised why they were not cleaned on the after scenario then and working now.

**Comment by amrita-shrestha** [comment]
> * In the issue `401` i.e. unauthorized is mentioned in the response status while cleaning in the after scenario. how does this change solves that case thinking
> * does this PR fixes the related issue? If yes, Still for a successful scenario like you mentioned here https://user-images.githubusercontent.com/54478846/182773375-1063ef25-5464-4d3a-970b-43375432ed80.webm. For this, the code was pushing the users to the created list right? There is no fail in the create request. so, I'm surprised why they were not cleaned on the after scenario then and working now.
  
@kiranparajuli589  We used to add users to the array before creating. In scenarios like user creation fails, the user-created array is updated but the user doesn't create so it stuck in cleanup step
https://github.com/owncloud/owncloud-test-middleware/blob/main/src/app.js#L120

**Comment by kiranparajuli589** [comment]
> > * In the issue `401` i.e. unauthorized is mentioned in the response status while cleaning in the after scenario. how does this change solves that case thinking
> > * does this PR fixes the related issue? If yes, Still for a successful scenario like you mentioned here https://user-images.githubusercontent.com/54478846/182773375-1063ef25-5464-4d3a-970b-43375432ed80.webm. For this, the code was pushing the users to the created list right? There is no fail in the create request. so, I'm surprised why they were not cleaned on the after scenario then and working now.
> 
> @kiranparajuli589 We used to add users to the array before creating. In scenarios like user creation fails, the user-created array is updated but the user doesn't create so it stuck in cleanup step https://github.com/owncloud/owncloud-test-middleware/blob/main/src/app.js#L120

Yes and that's great! We shouldn't be deleting non-existing users. :partying_face: 
Now does this fix the case you shared in that video?

**Comment by amrita-shrestha** [comment]
> > > * In the issue `401` i.e. unauthorized is mentioned in the response status while cleaning in the after scenario. how does this change solves that case thinking
> > > * does this PR fixes the related issue? If yes, Still for a successful scenario like you mentioned here https://user-images.githubusercontent.com/54478846/182773375-1063ef25-5464-4d3a-970b-43375432ed80.webm. For this, the code was pushing the users to the created list right? There is no fail in the create request. so, I'm surprised why they were not cleaned on the after scenario then and working now.
> > 
> > 
> > @kiranparajuli589 We used to add users to the array before creating. In scenarios like user creation fails, the user-created array is updated but the user doesn't create so it stuck in cleanup step https://github.com/owncloud/owncloud-test-middleware/blob/main/src/app.js#L120
> 
> Yes and that's great! We shouldn't be deleting non-existing users. partying_face Now does this fix the case you shared in that video?

video shows the cleanup is present in the after scenario. @kiranparajuli589  can you come online in rocket chat?

File: src/helpers/graphHelper.js (hunk 1 above)
Comment (amrita-shrestha) [reply src/helpers/graphHelper.js:31]: @kiranparajuli589 in my view order doesn't matter because we res.ok check the response and only add user in array

**Review by kiranparajuli589 — APPROVED** [review]

**Comment by amrita-shrestha** [comment]
> rest lgtm +1 the `401` case reported in the issue should be fixed with this as non-existing user are now not in the created list.

do we need that because this change only adds a user in  user created array after successful creation

**Comment by amrita-shrestha** [comment]
> rest lgtm +1 the `401` case reported in the issue should be fixed with this as non-existing user are now not in the created list.

@kiranparajuli589 i had refactored the error message so that its easy to find out.

**Comment by kiranparajuli589** [comment]
@amrita-shrestha please do the release (minor is enough) and bumps followups.

File: src/helpers/sharingHelper.js [hunk 2]
```diff
@@ -253,7 +253,7 @@ module.exports = {
     return httpHelper
       .getOCS(apiURL, user)
       .then((res) => {
-        httpHelper.checkStatus(res, 'The response status is not the expected value')
+        httpHelper.checkStatus(res, 'The HTTP response status is not the expected value in response to getting all shares.')
```
Comment (saw-jan) [inline src/helpers/sharingHelper.js:256]: I am confused by this message!
```suggestion
        httpHelper.checkStatus(res, 'The HTTP response status is not the expected value')
```

Comment (amrita-shrestha) [reply src/helpers/sharingHelper.js:256]: @sawjan dai **The HTTP response status is not the expected value**  this message is the same for all. Hard to differentiate the request type

**Pull request merged**